backend/audit/models/models.py

```diff
@@ -527,3 +529,55 @@ def get_transition_date(self, status):
         if indices:
             return self.transition_date[indices[-1]]
         return None
+
+    # Resubmission SAC Creations
+    # Atomically create a new SAC row as a resubmission of this SAC. Assert that a resubmission does not already exist
+    def initiate_resubmission(self, user=None, event_type=None):
+        with transaction.atomic():
+            existing_resub = SingleAuditChecklist.objects.filter(
+                resubmission_meta__previous_report_id=self.report_id
+            ).exists()
+
+            if existing_resub:
+                raise ValidationError(
+                    f"A resubmission already exists for report_id {self.report_id}."
+                )
+
+            resub = SingleAuditChecklist.objects.create(
```

Comment:
Is there a way to copy the `self` object that is less cumbersome? (I have no idea.) Would `copy.deepcopy` work on a Django object? (If so... would that work? Would it... have unintended consequences?) I only ask because this looks fragile to maintain... we have to remember all the fields of the `sac` object, and any time anything changes, we have to update this. Hence, I thought I'd ask. :shrug: